### PR TITLE
Event capabilities

### PR DESCRIPTION
**Pull request by psrpinto**
This PR introduces the following custom capabilities, and changes the whole codebase so that these capabilities are used, through calls to `current_user_can()`:

- `create_translation_event`
- `view_translation_event`
- `edit_translation_event`
- `delete_translation_event`

There should be no changes to how things work, everything should work the same as before.

## References

- [Meta capabilities for custom post types](https://justintadlock.com/archives/2010/07/10/meta-capabilities-for-custom-post-types)
- [`current_user_can()`](https://developer.wordpress.org/reference/functions/current_user_can/)
- [`map_meta_cap()`](https://developer.wordpress.org/reference/hooks/map_meta_cap/)

> This function also accepts an ID of an object to check against if the capability is a meta capability. Meta capabilities such as edit_post and edit_user are capabilities used by the map_meta_cap() function to map to primitive capabilities that a user or role has, such as edit_posts and edit_others_posts.

- [`friends/includes/class-access-control.php#L52`](https://github.com/akirk/friends/blob/5a9ee0d07c021ef1bd022790e4ce9cfb4b34d161/includes/class-access-control.php#L52)
- [`friends/includes/class-access-control.php#L302`](https://github.com/akirk/friends/blob/5a9ee0d07c021ef1bd022790e4ce9cfb4b34d161/includes/class-access-control.php#L302)
- [`global.wordpress.org/public_html/wp-content/mu-plugins/roles/cross-locale-pte.php#L386`](https://github.com/WordPress/wordpress.org/blame/trunk/global.wordpress.org/public_html/wp-content/mu-plugins/roles/cross-locale-pte.php#L386)

### CODE REVIEW

**Review by akirk — APPROVED** [review]
Great job, this vastly improves the readability and ensures consistency. Also thanks for the tests!

**Review by akirk — CHANGES_REQUESTED** [review]
I have found one thing that no longer works: When an event doesn't have a host, the creator should be able to edit.

File: includes/event/event-capabilities.php
Comment (akirk) [inline includes/event/event-capabilities.php]: Since we're returning, we can remove the `else`.
```suggestion
				}
				if ( self::EDIT === $cap ) {
```

File: includes/event/event-capabilities.php
Comment (akirk) [inline includes/event/event-capabilities.php:118]: Maybe this is too early and/or we need more fine grained controls (which is easy now!) But I think it should be possible to change at least the event description (and maybe the event end, if it should be extended) even when there are stats.

**Review by akirk — APPROVED** [review]
I'm going to re-approve this. I think this might not change the current behavior but we might need to discuss it wrt when editing an event should remain possible.

File: includes/event/event-capabilities.php
Comment (psrpinto) [reply includes/event/event-capabilities.php:118]: Indeed, more fine grained capabilities could make sense, e.g. `edit_translation_event_with_stats`. Also agree that it it should be possible to edit certain fields of a past event. I opened https://github.com/WordPress/wporg-gp-translation-events/issues/227 to track that.

**Comment by psrpinto** [comment]
> I have found one thing that no longer works: When an event doesn't have a host, the creator should be able to edit.

I'm working on a fix for this

**Comment by akirk** [comment]
@psrpinto I'm no longer sure this is true. I think I had tested with an event that already had contributions.

**Comment by psrpinto** [comment]
There was indeed an issue when the event_id is a `string` and not an `int`. Now fixed.

**Comment by psrpinto** [comment]
Merging since @akirk had approved, and after his approval there was only the bug fix in the last commit.

**Comment by akirk** [comment]
Sorry for missing the re-review, thanks!

**Comment by psrpinto** [comment]
No worries!

**Pull request merged**
